executor/analyze.go

```diff
@@ -206,9 +211,70 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
 			}
 		}
 	}
+	err = e.saveAnalyzeOptsV2()
+	if err != nil {
+		e.ctx.GetSessionVars().StmtCtx.AppendWarning(err)
+	}
 	return statsHandle.Update(e.ctx.GetInfoSchema().(infoschema.InfoSchema))
 }
 
+func (e *AnalyzeExec) saveAnalyzeOptsV2() error {
+	if !variable.PersistAnalyzeOptions.Load() || e.V2AnalyzeOptions == nil {
+		return nil
+	}
+	err := saveAnalyzeOpts(e.V2AnalyzeOptions, e.ctx.(sqlexec.RestrictedSQLExecutor))
+	if err != nil {
+		return err
+	}
+	for _, task := range e.tasks {
+		if task.colExec != nil && task.colExec.PhyTableID != e.PhyTableID {
+			err = saveAnalyzeOpts(task.colExec.V2AnalyzeOptions, e.ctx.(sqlexec.RestrictedSQLExecutor))
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+var colChoiceEnum = [4]string{"UNDEFINED", "ALL", "PREDICATE", "LIST"}
+
+func saveAnalyzeOpts(opts *core.V2AnalyzeOptions, exec sqlexec.RestrictedSQLExecutor) error {
+	sampleNum := uint64(0)
+	if val, ok := opts.RawOpts[ast.AnalyzeOptNumSamples]; ok {
+		sampleNum = val
+	}
+	sampleRate := float64(0)
+	if val, ok := opts.RawOpts[ast.AnalyzeOptSampleRate]; ok {
+		sampleRate = math.Float64frombits(val)
+	}
+	buckets := uint64(0)
+	if val, ok := opts.RawOpts[ast.AnalyzeOptNumBuckets]; ok {
+		buckets = val
+	}
+	topn := uint64(0)
+	if val, ok := opts.RawOpts[ast.AnalyzeOptNumTopN]; ok {
+		topn = val
+	}
+	colChoice := colChoiceEnum[opts.ColChoice]
```

Comment:
Can we use `opts.ColChoice.String()` here?

Reply:
Or unify model.ColumnChoice.String with mysql.analyze_options.cholumn_choice?

Reply:
Good suggestion, aligned strings of ColChoice.String()